aptos-move/aptos-vm/src/parallel_executor/mod.rs

```diff
@@ -86,57 +88,98 @@ impl ParallelAptosVM {
         transactions: Vec<Transaction>,
         state_view: &S,
         concurrency_level: usize,
-    ) -> Result<(Vec<TransactionOutput>, Option<Error<VMStatus>>), VMStatus> {
+    ) -> Result<Vec<TransactionOutput>, VMStatus> {
         // Verify the signatures of all the transactions in parallel.
         // This is time consuming so don't wait and do the checking
         // sequentially while executing the transactions.
-        let signature_verified_block: Vec<PreprocessedTransaction> = transactions
-            .par_iter()
-            .map(|txn| preprocess_transaction::<AptosVM>(txn.clone()))
-            .collect();
-
-        match ParallelTransactionExecutor::<PreprocessedTransaction, AptosVMWrapper<S>>::new(
-            concurrency_level,
-        )
-        .execute_transactions_parallel(state_view, signature_verified_block)
-        {
-            Ok((results, delta_resolver)) => {
-                // TODO: with more deltas, collect keys in parallel (in parallel executor).
-                let mut aggregator_keys: HashMap<StateKey, anyhow::Result<ResolvedData>> =
-                    HashMap::new();
-
-                for res in results.iter() {
-                    let output_ext = AptosTransactionOutput::as_ref(res);
-                    for (key, _) in output_ext.delta_change_set().iter() {
-                        if !aggregator_keys.contains_key(key) {
-                            aggregator_keys.insert(key.clone(), state_view.get_state_value(key));
+        // TODO: use the same threadpool as for parallel execution.
+        let signature_verified_block: Vec<PreprocessedTransaction> =
+            RAYON_EXEC_POOL.install(|| {
+                transactions
+                    .par_iter()
+                    .map(|txn| preprocess_transaction::<AptosVM>(txn.clone()))
+                    .collect()
+            });
+
+        let log_context = AdapterLogSchema::new(state_view.id(), 0);
+        info!(
+            log_context,
+            "Executing block, transaction count: {}",
+            transactions.len()
+        );
+
+        let executor =
+            ParallelTransactionExecutor::<PreprocessedTransaction, AptosVMWrapper<S>>::new(
+                concurrency_level,
+            );
+
+        let mut ret = None;
+        if concurrency_level > 1 {
+            ret = Some(
```

Comment:
there's too many indentation here, I'd also try to extract the delta handling out from this block

Reply:
refactored the whole thing